### PR TITLE
Docstring convention update: colons good, types in typehints

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -196,32 +196,32 @@ https://numpydoc.readthedocs.io/en/latest/format.html). An example is shown belo
...
-  kappa
+  kappa:
       Plasma elongation
 
   Returns
   -------
-  tau_E, the energy confinement time [s]
-
+  tau_E:
+      Energy confinement time [s]
```
Comment (je-cook) [inline CONTRIBUTING.md:224]: tau_E will be seen as the type I think? Need to check the sphinx docs
```suggestion
      Energy confinement time [s]
```

**Review by je-cook — COMMENTED** [review]
Just the one comment

File: CONTRIBUTING.md (hunk 1 above)
Comment (CoronelBuendia) [reply CONTRIBUTING.md:224]: bugger, you're right

**Comment by tplobo** [comment]
This new version of `CONTRIBUTING.md` looks really good. It was a read that I was truly missing when I started!

**Pull request merged**